system/metrics/sinks/elastic/elasticSink.hpp

```diff
@@ -0,0 +1,59 @@
+/*##############################################################################
+    HPCC SYSTEMS software Copyright (C) 2021 HPCC Systems®.
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+
+#pragma once
+
+#include "jmetrics.hpp"
+#include "jptree.hpp"
+#include "jstring.hpp"
+
+//including cpp-httplib single header file REST client
+//  doesn't work with format-nonliteral as an error
+//
+#if defined(__clang__) || defined(__GNUC__)
+#pragma GCC diagnostic push
```

Comment:
are all of these necessary?

Reply:
If not included, the following build errors are generated:

In file included from /home/krowland/hpcc/src/system/metrics/sinks/elastic/elasticSink.hpp:30,
                 from /home/krowland/hpcc/src/system/metrics/sinks/elastic/elasticSink.cpp:14:
/home/krowland/hpcc/src/system/httplib/httplib.h: In instantiation of ‘ssize_t httplib::Stream::write_format(const char*, const Args& ...) [with Args = {int, const char*}; ssize_t = long int]’:
/home/krowland/hpcc/src/system/httplib/httplib.h:3940:61:   required from here
/home/krowland/hpcc/src/system/httplib/httplib.h:3635:21: error: format not a string literal, argument types not checked [-Werror=format-nonliteral]
 3635 |   auto sn = snprintf(buf.data(), buf.size() - 1, fmt, args...);
      |             ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/krowland/hpcc/src/system/httplib/httplib.h:3652:19: error: format not a string literal, argument types not checked [-Werror=format-nonliteral]
 3652 |           snprintf(&glowable_buf[0], glowable_buf.size() - 1, fmt, args...));
      |           ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/krowland/hpcc/src/system/httplib/httplib.h: In instantiation of ‘ssize_t httplib::Stream::write_format(const char*, const Args& ...) [with Args = {const char*, const char*}; ssize_t = long int]’:
/home/krowland/hpcc/src/system/httplib/httplib.h:4805:76:   required from here
/home/krowland/hpcc/src/system/httplib/httplib.h:3635:21: error: format not a string literal, argument types not checked [-Werror=format-nonliteral]
 3635 |   auto sn = snprintf(buf.data(), buf.size() - 1, fmt, args...);
      |             ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/krowland/hpcc/src/system/httplib/httplib.h:3652:19: error: format not a string literal, argument types not checked [-Werror=format-nonliteral]
 3652 |           snprintf(&glowable_buf[0], glowable_buf.size() - 1, fmt, args...));
      |           ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

There is a similar error w/o the #undef for INVALID_SOCKET